### PR TITLE
feat: add blockHash to verificationClaim

### PR DESCRIPTION
**Pull request by zachterrell57**
## Motivation

Ordering by blockHash allows applications to resolve conflicts easily, and lets users nullify older verifications by issuing new ones with a more recent blockhash

## Change Type

_Choose one of the options below by adding an `x` now or at any time before submitting for review_

- [ ] Protocol Feature: changes the design of the hub and the protocol
- [x] Hub Feature: changes the design of the hub, but does not affect the protocol
- [ ] Bugfix: solves an issue with the current implementation
- [ ] Chore: miscellaneous improvement to the system

## Change Summary

Added blockHash to both VerificationClaim type and VerificationAddBody type. Updated tests and `engine` accordingly to account for this new property.

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [x] The title of this PR adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [ ] Changes to the protocol specification have been merged

## Additional Context

If this is a relatively large or complex change, provide more details here that will help reviewers

### CODE REVIEW

File: src/types/index.ts [hunk 1]
```diff
@@ -150,13 +150,15 @@ export type VerificationAdd = Message<VerificationAddBody>;
  *
  * @externalUri - the URI of the external entity
  * @claimHash - the hash of the verification claim.
+ * @blockHash - the block hash of the current block
  * @externalSignature - the signature of the hash of the verification claim, signed by the external key pair.
- * @externalSignatureType - type of signature from set of supported types (see version 0x45 of https://eips.ethereum.org/EIPS/eip-191 for 'eip-191-0x45')
+ * @externalSignatureType - type of signature from set of supported types (see version 0x45 of https://eips.ethereum.org/EIPS/eip-191 for 'eip-191-0x45').
```
Comment (pfletcherhill) [inline src/types/index.ts:155]: Thought we didn't want periods after comments

File: src/factories.ts [hunk 2]
```diff
@@ -206,6 +207,7 @@ export const Factories = {
           body: {
             externalUri: ethWallet.address,
             claimHash: '',
+            blockHash: '',
```
Comment (pfletcherhill) [inline src/factories.ts:210]: This should be faked (i.e. `Faker.datatype.hexaDecimal(64)`)

File: src/factories.ts [hunk 3]
```diff
@@ -234,6 +236,7 @@ export const Factories = {
           const verificationClaim: VerificationClaim = {
             username: props.data.username,
             externalUri,
+            blockHash: Faker.datatype.hexaDecimal(64).toLowerCase(),
```
Comment (pfletcherhill) [inline src/factories.ts:239]: What's the reason to include `toLowerCase`?

File: src/engine/engine.verification.test.ts [hunk 4]
```diff
@@ -33,10 +34,12 @@ describe('mergeVerification', () => {
     aliceAddress = aliceSigner.signerKey;
     transientParams = { transient: { signer: aliceSigner, ethWallet: aliceEthWallet } };
     aliceRoot = await Factories.Root.create({ data: { rootBlock: 100, username: 'alice' } }, transientParams);
+    aliceBlockHash = Faker.datatype.hexaDecimal(64).toLowerCase();
```
Comment (pfletcherhill) [inline src/engine/engine.verification.test.ts:37]: Why `toLowerCase`?

**Review by pfletcherhill — COMMENTED** [review]
Couple minor comments

File: src/types/index.ts (hunk 1 above)
Comment (zachterrell57) [reply src/types/index.ts:155]: forgot that one, should I just go through and un-punctuate all the comments in this class?

File: src/factories.ts (hunk 3 above)
Comment (zachterrell57) [reply src/factories.ts:239]: when we Fake a blockHash in Factory.Root it uses toLowerCase(), just copied that here

File: src/engine/engine.verification.test.ts (hunk 4 above)
Comment (zachterrell57) [reply src/engine/engine.verification.test.ts:37]: see above

File: src/types/index.ts (hunk 1 above)
Comment (pfletcherhill) [reply src/types/index.ts:155]: Sure, no harm in doing that!

File: src/factories.ts (hunk 3 above)
Comment (pfletcherhill) [reply src/factories.ts:239]: Okay, a block hash should be case insensitive, so doing `toLowerCase()` doesn't really matter. But maybe it makes it look more consistent with block hashes that come from other libraries that are lowercased. I'm fine either way.

**Comment by pfletcherhill** [comment]
@zachterrell57 can you pull from origin/main and resolve conflicts? Sorry, I merged a bunch of signer set code earlier.

**Review by pfletcherhill — APPROVED** [review]
Lgtm

**Pull request merged**